CMakeLists.txt

```diff
@@ -302,50 +337,102 @@ endif()
-    Luau.CodeGen 
-    Luau.Compiler 
-    Luau.VM 
-    Luau.Analysis 
-    onig 
-    plutovg 
-    tinyxml2 
-    DearImGui
-    # FFmpeg stuff
-    ${LIB_AVCODEC}
-    ${LIB_AVDEVICE}
-    ${LIB_AVFILTER}
-    ${LIB_AVFORMAT}
-    ${LIB_AVUTIL}
-    ${LIB_SWRESAMPLE}
-    ${LIB_SWSCALE}
-    # Windows static libs required for ffmepg
-    Ws2_32
-    Secur32
-    Bcrypt
-    Mfuuid
-    Strmiids
-)
+if (MSVC)
+    find_library(LIB_AVCODEC libavcodec HINTS ${CMAKE_SOURCE_DIR}/Animations/vendor/ffmpeg/build/lib)
+    find_library(LIB_AVDEVICE libavdevice HINTS ${CMAKE_SOURCE_DIR}/Animations/vendor/ffmpeg/build/lib)
+    find_library(LIB_AVFILTER libavfilter HINTS ${CMAKE_SOURCE_DIR}/Animations/vendor/ffmpeg/build/lib)
+    find_library(LIB_AVFORMAT libavformat HINTS ${CMAKE_SOURCE_DIR}/Animations/vendor/ffmpeg/build/lib)
+    find_library(LIB_AVUTIL libavutil HINTS ${CMAKE_SOURCE_DIR}/Animations/vendor/ffmpeg/build/lib)
+    find_library(LIB_SWRESAMPLE libswresample HINTS ${CMAKE_SOURCE_DIR}/Animations/vendor/ffmpeg/build/lib)
+    find_library(LIB_SWSCALE libswscale HINTS ${CMAKE_SOURCE_DIR}/Animations/vendor/ffmpeg/build/lib)
+else()
+    find_library(LIB_AVCODEC avcodec HINTS ${CMAKE_SOURCE_DIR}/Animations/vendor/ffmpeg/build/lib)
+    find_library(LIB_AVDEVICE avdevice HINTS ${CMAKE_SOURCE_DIR}/Animations/vendor/ffmpeg/build/lib)
+    find_library(LIB_AVFILTER avfilter HINTS ${CMAKE_SOURCE_DIR}/Animations/vendor/ffmpeg/build/lib)
+    find_library(LIB_AVFORMAT avformat HINTS ${CMAKE_SOURCE_DIR}/Animations/vendor/ffmpeg/build/lib)
+    find_library(LIB_AVUTIL avutil HINTS ${CMAKE_SOURCE_DIR}/Animations/vendor/ffmpeg/build/lib)
+    find_library(LIB_SWRESAMPLE swresample HINTS ${CMAKE_SOURCE_DIR}/Animations/vendor/ffmpeg/build/lib)
+    find_library(LIB_SWSCALE swscale HINTS ${CMAKE_SOURCE_DIR}/Animations/vendor/ffmpeg/build/lib)
+endif()
+
+if(WIN32)
+    target_link_libraries(
+        MathAnimations
+        glfw
+        freetype
+        DearImGuiBackend
+        Glad
+        NativeFileDialog
+        OpenAL
+        Luau.Ast
+        Luau.CodeGen
+        Luau.Compiler
+        Luau.VM
+        Luau.Analysis
+        onig
+        plutovg
+        tinyxml2
+        DearImGui
+        # FFmpeg stuff
+        ${LIB_AVCODEC}
+        ${LIB_AVDEVICE}
+        ${LIB_AVFILTER}
+        ${LIB_AVFORMAT}
+        ${LIB_AVUTIL}
+        ${LIB_SWRESAMPLE}
+        ${LIB_SWSCALE}
+        # Windows static libs required for ffmepg
+        Ws2_32
+        Secur32
+        Bcrypt
+        Mfuuid
+        Strmiids
+    )
+else()
+    target_link_libraries(
+        MathAnimations
+
+        ssl
+        crypto
+
+        glfw
+        freetype
+        DearImGuiBackend
+        Glad
+        NativeFileDialog
+        OpenAL
+        Luau.Ast
+        Luau.CodeGen
+        Luau.Compiler
+        Luau.VM
+        Luau.Analysis
+        onig
+        plutovg
+        tinyxml2
+        DearImGui
+        # FFmpeg stuff
+        va
+        vdpau
```

Comment:
vdpau and friends are not needed yet, and checks if the needed library and header files is present is not performed either.

Example:
Animations/vendor/ffmpeg/libavutil/hwcontext.c only references vdpau code if CONFIG_VAAPI is defined

Reply:
On my system, this is automatically defined.
Getting these libraries dynamically is a TODO